lib/ecl/fortio.c

```diff
@@ -119,7 +119,7 @@ static fortio_type * fortio_alloc__(const char *filename , bool fmt_file , bool
 */
 
 static bool __read_int(FILE * stream , int * value, bool endian_flip) {
-    return !eclfio_sizeof( stream, endian_flip ? "" : "e", value );
+    return !eclfio_sizeof( stream, endian_flip ? "c" : "ce", value );
```

Comment:
I assume this should have the `"i"` type specifier, otherwise the endian swapping of the data will not be correct?

Reply:
I thought so too (which is why the tests failed), turns out __read_int expects the size in bytes of the current record, not the number-of-integers, so it always needs `c`. `eclfio_sizeof` doesn't swap bytes except for the header, which is always assumed to be `int32`